app/controllers/messages_controller.rb

```diff
@@ -17,7 +17,8 @@ def index
 
   def new
     @message = Message.new
-    @message.recipient = User.find_by_id(params[:recipient_id])
+    recipient = User.find_by_id(params[:recipient_id])
+    @message.recipient = recipient if recipient&.can_interact_with?(current_user)
```

Comment:
I'm finding following all the different method names to be surprisingly hard; that said, I don't think we should strip users unless the current_user has blocked the recipient, and I think you strip them both ways?

Reply:
This is still both ways